compiler/rustc_hir_analysis/src/collect/lifetimes.rs

```diff
@@ -1329,6 +1333,58 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
         );
     }
 
+    fn resolve_late_type_ref(&mut self, param_def_id: LocalDefId, hir_id: hir::HirId) {
```

Comment:
Can this be merged with the logic for lifetimes?

Reply:
I kept the logic separate because this doesn't need all the error reporting logic as the other one -- I could merge them if you'd really like, though.